File: Day-08/Single_Number.java
import java.util.*;
class Solution {
    
    static int singleNumber(int arr[]){

        int sum=0;
        for(int i=0; i<arr.length; i++){
             

            sum = sum ^arr[i];
        }
        return sum;
    }

    public static void main(String[] args){

        int test_case1[] = new int[] {2,2,1};
        int test_case2[] = new int[] {4,1,2,1,2};
        int test_case3[] = new int[] {1};

        int output1 = singleNumber(test_case1);
        int output2= singleNumber(test_case2);
        int output3= singleNumber(test_case3);

        System.out.println(Arrays.toString(test_case1) + " Output is "+output1);
        System.out.println(Arrays.toString(test_case2) + " Output is "+output2);
        System.out.println(Arrays.toString(test_case3) + " Output is "+output3);
    }
}
    

File: Day-08/Single_number.py
def Single_num(arr):
    num=0
    for i in range(0,len(arr)):
        num=num^arr[i]
    return num

test_case_1 = [2,2,1]
test_case_2 = [4,1,2,1,2]
test_case_3 = [1]

print("--- test case 1 ---")
print(f"Input: nums = {test_case_1}\nOutput: {Single_num(test_case_1)}")
print("--- test case 2 ---")
print(f"Input: nums = {test_case_2}\nOutput: {Single_num(test_case_2)}")
print("--- test case 3 ---")
print(f"Input: nums = {test_case_3}\nOutput: {Single_num(test_case_3)}")